source/navitiacommon/navitiacommon/models.py

```diff
@@ -329,10 +330,23 @@ def last_datasets(self, nb_dataset=1):
         return result
 
     @classmethod
+    def query_existing(cls):
+        return cls.query.filter_by(discarded=False)
+
+    @classmethod
     def get_by_name(cls, name):
-        res = cls.query.filter_by(name=name).first()
+        res = cls.query_existing().filter_by(name=name).first()
         return res
 
+    @classmethod
+    def get_from_id_or_name(cls, id=None, name=None):
+        if id:
+            return cls.query.get_or_404(id)
+        elif name:
+            return cls.query_existing().filter_by(name=name).first_or_404()
+        else:
+            return {'error': 'instance is required'}, 400
```

Comment:
is this done in other API ? I'm not a fan of the fact that you need to explicitly get returned tuple from the calling function and then treat it as an error.

wouldn't it be better to throw an exception (an exception that will end with a 404) ?

@kinnou02 what do you think ?